src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediatePreparedStatement.java

```diff
@@ -102,7 +102,7 @@ public void execute() {
     this.executed = true;
     try {
       ((PreparedStatement) this.statement).execute();
-      this.executeHelper();
+      this.updateResultCount(null);
```

Comment:
not a fan of null.. can we either create a new method definition that doesnt take input or maybe use empty type ?

Reply:
I refactored this into a batch and non-batch version.